src/ucs/stats/stats.c

```diff
@@ -195,6 +311,9 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
         ucs_stats_node_remove(node, 1);
     } else {
         ucs_stats_node_remove(node, 0);
+        if (!node->filter_node->type_list_len) {
```

Comment:
looks like this code block should go into ucs_stats_node_remove(node, 0) - since same thing is done below in line 519